sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1715,6 +1715,22 @@ static uint32_t _handle_rio(gnrc_netif_t *netif, const ipv6_hdr_t *ipv6,
                              netif->pid, route_ltime);
     }
 
+    if (IS_ACTIVE(CONFIG_GNRC_NETIF_IPV6_BR_AUTO_6CTX) && gnrc_netif_is_6lbr(netif)) {
+        /* configure compression context */
+        int cid = gnrc_sixlowpan_ctx_update_6ctx(&rio->prefix, rio->prefix_len, route_ltime);
+        if (cid >= 0) {
+            DEBUG("gnrc_netif: add compression context for prefix %s/%u\n",
+                   ipv6_addr_to_str(addr_str, &rio->prefix, sizeof(addr_str)),
+                   rio->prefix_len);
+
+            /* add new context to ABRs */
+            _nib_abr_entry_t *abr = NULL;
+            while ((abr = _nib_abr_iter(abr))) {
+                bf_set(abr->ctxs, cid);
+            }
```

Comment:
Not sure if the context should be associated to all stored BRs or just to us.

Reply:
By now I would say just to us.

Reply:
Is there an easy way to find out which one we are?

Reply:
cycle through the ABRs and check if `netif` has assigned the BR address I would say

Reply:
`_handle_abro()` is called before `_handle_rio` so the ABR should be there

Reply:
> A router processes RS messages as specified in [[RFC4861](https://www.rfc-editor.org/rfc/rfc4861)].  The
>   differences relate to the inclusion of ABROs in the RA messages and
>   the exclusive use of unicast RAs.  If a 6LR has received an ABRO from
>  a 6LBR, then it will include that option unmodified in the RA
>   messages it sends.  And, if the 6LR has received RAs -- whether with
>   the same prefixes and context information or different -- from a
>   different 6LBR, then it will need to keep those prefixes and that
>   context information separately so that the RAs the 6LR sends will
>   maintain the association between the ABRO and the prefixes and
>   context information.  The router can tell which 6LBR originated the
>   prefixes and context information from the 6LBR Address field in the
>   ABRO.

Therefor the bitfield association